cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -1108,6 +1108,28 @@ pub(crate) fn define(
         .operands_out(vec![a]),
     );
 
+    let mask = &operand_doc(
+        "mask",
+        &imm.uimm128,
+        "The 16 immediate bytes used for selecting the elements to shuffle",
+    );
+    let a = &operand_doc("a", TxN, "A vector value");
+    let b = &operand_doc("b", TxN, "A vector value");
+
+    ig.push(
+        Inst::new(
+            "shuffle",
+            r#"
+        SIMD vector shuffle.
+        Shuffle two vectors using the given immediate bytes. For each of the 16 bytes of the
+        immediate, a value i of 0-15 selects the i-th element of the first vector and a value i of 
```

Comment:
I ended up restricting the type for this to only `Tx16` types (which turns out can only be ints and floats), which forces a `raw_bitcast` to shuffle another type. I think it would be nice to have a more general `shuffle` but right now the WASM SIMD spec only allows `v8x16`.